File: app/src/main/java/com/forkmang/adapter/ViewPagerAdapter_SelectFood.java
package com.forkmang.adapter;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.lifecycle.Lifecycle;
import androidx.viewpager2.adapter.FragmentStateAdapter;

import com.forkmang.data.FoodList_Tab;
import com.forkmang.fragment.Select_Food_Fragment;
import com.forkmang.data.RestoData;
import com.forkmang.models.TableList;

import java.util.ArrayList;

public class ViewPagerAdapter_SelectFood extends FragmentStateAdapter {
    ArrayList<FoodList_Tab> foodListArrayList;
    TableList tableList;
    RestoData restoData;

    public ViewPagerAdapter_SelectFood(@NonNull FragmentManager fragmentManager, @NonNull Lifecycle lifecycle, ArrayList<FoodList_Tab> foodListArrayList, TableList tableList, RestoData restoData) {
        super(fragmentManager, lifecycle);
        this.foodListArrayList = foodListArrayList;
        this.tableList = tableList;
        this.restoData = restoData;
    }

    @NonNull
    @Override
    public Fragment createFragment(int position) {
        Fragment selectedFragment;
        selectedFragment= Select_Food_Fragment.newInstance(tableList, restoData);
        return  selectedFragment;
    }

    @Override
    public int getItemCount() {
        return foodListArrayList.size();
    }
}

File: app/src/main/java/com/forkmang/adapter/ViewPagerAdapter_SelectFood.kt
package com.forkmang.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.lifecycle.Lifecycle
import androidx.viewpager2.adapter.FragmentStateAdapter
import com.forkmang.data.FoodList_Tab
import com.forkmang.data.RestoData
import com.forkmang.fragment.Select_Food_Fragment
import com.forkmang.models.TableList

class ViewPagerAdapter_SelectFood(
    fragmentManager: FragmentManager,
    lifecycle: Lifecycle,
    var foodListArrayList: ArrayList<FoodList_Tab>,
    var tableList: TableList,
    var restoData: RestoData
) : FragmentStateAdapter(fragmentManager, lifecycle) {
    override fun createFragment(position: Int): Fragment {
        val selectedFragment: Fragment
        selectedFragment = Select_Food_Fragment.newInstance(tableList, restoData)
        return selectedFragment
    }

    override fun getItemCount(): Int {
        return foodListArrayList.size
    }
}